nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java

```diff
@@ -292,7 +296,35 @@ private static boolean isDirectTypeUseAnnotation(Attribute.TypeCompound t) {
     // location is a list of TypePathEntry objects, indicating whether the annotation is
     // on an array, inner type, wildcard, or type argument. If it's empty, then the
     // annotation is directly on the type.
-    return t.position.location.isEmpty();
+    // We care about both annotations directly on the outer type and also those directly
+    // on an inner type or array dimension, but wish to discard annotations on wildcards,
+    // or type arguments.
+    // For arrays, we treat annotations on the outer type and on any dimension of the array
+    // as applying to the nullability of the array itself, not the elements.
+    // We don't allow mixing of inner types and array dimensions in the same location
+    // (i.e. `Foo.@Nullable Bar []` is meaningless).
+    // These aren't correct semantics for type use annotations, but a series of hacky
+    // compromises to keep some semblance of backwards compatibility until we can do a
+    // proper deprecation of the incorrect behaviors for type use annotations when their
+    // semantics don't match those of a declaration annotation in the same position.
+    // See https://github.com/uber/NullAway/issues/708
+    boolean location_has_inner_types = false;
```

Comment:
Stick with standard lowerCamelCase?
```suggestion
    boolean locationHasInnerTypes = false;
```

Reply:
Ugh, sorry, the perils of alt-tabbing between Python and Java code 😅